### ISSUE TITLE
Blank screen on designer page - Failed to load resource

### ISSUE THREAD

**Issue by maheini** [frappe/print_designer#325]
Behavior:
Print designer form popup is visible, but after selecting a template and defining it's name, print designer will just show a white screen (except for the header).

Expected behavior: 
Print designer canvas should be visible

<details>
<summary>Environment</summary>
<p>
<br> 
Frappe: v15.33.3<br>
ERPNext: v15.29.2 prod<br>
Print Designer: v1.4.0 (current main branch)<br>
Install-method: Docker, custom built (no traefik, no SSL, direct 8080 port access) image running in a local network
</p>
</details>

<details>
<summary>JS Console errors</summary>
<br>
<img src="https://github.com/frappe/print_designer/assets/65506676/28919a44-1c24-43e1-81f8-033a1c6a4273"/>
</details>

<details>
<summary>Apps.json content (custom build)</summary>
<p>
<br> 

```

[
    {
        "url": "https://github.com/frappe/payments",
        "branch": "version-15"
    },
    {
        "url": "https://github.com/frappe/erpnext",
        "branch": "version-15"
    },
    {
        "url": "https://github.com/frappe/print_designer.git",
        "branch": "main"
    }
]

```

</p>
</details>

<details>
<summary>Further details</summary>
<br>
<p>
Docker custom build is working perfectly fine without any issues for other apps.
</p>
</details>

**Comment by maharshivpatel** [comment]
can you bench build, migrate and restart bench.

**Comment by maheini** [comment]
Hi @maharshivpatel 
Thank you. I run those commands:
```
bench build
bench --site all migrate
bench restart
```
Then got this error message whilst trying to restart bench:
```
bench restart
WARN: restart failed: Couldn't find supervisorctl in PATH
```
Installed supervisor - and got a new error after trying to re-run bench restart:
<details>
  <summary>Error message details</summary>

```
root@debianserver:~# docker compose -p myproject exec backend bench restart
$ supervisorctl restart frappe:
unix:///var/run/supervisor.sock no such file
ERROR: supervisorctl restart frappe:
subprocess.CalledProcessError: Command 'supervisorctl restart frappe:' returned non-zero exit status 7.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/bench", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/usr/local/lib/python3.11/site-packages/bench/cli.py", line 132, in cli
    bench_command()
  File "/usr/local/lib/python3.11/site-packages/bench/commands/utils.py", line 41, in restart
    Bench(".").reload(web, supervisor, systemd)
  File "/usr/local/lib/python3.11/site-packages/bench/utils/render.py", line 126, in wrapper_fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/bench/bench.py", line 152, in reload
    restart_supervisor_processes(bench_path=self.name, web_workers=web, _raise=_raise)
  File "/usr/local/lib/python3.11/site-packages/bench/utils/bench.py", line 342, in restart_supervisor_processes
    failure = bench.run(f"{sudo}supervisorctl restart {group}", _raise=_raise)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/bench/bench.py", line 48, in run
    return exec_cmd(cmd, cwd=cwd or self.cwd, _raise=_raise)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/bench/utils/__init__.py", line 178, in exec_cmd
    raise CommandFailedError(cmd) from subprocess.CalledProcessError(return_code, cmd)
bench.exceptions.CommandFailedError: supervisorctl restart frappe:
```
</details>

Restarted my docker containers and checked the page.... but it seems like all assets are messed up now. CSS isn't loading anymore on ERPNext.

**Comment by maharshivpatel** [comment]
@maheini please go/search though discuss.frappe.io and post your question there to find solutions regarding your setup. There are people who faced similar issue and can help you. As this is not related to print_designer closing this for now. Thank you.

**Comment by maheini** [comment]
@maharshivpatel I think there might be a good chance it's actually a print_designer issue rather than a frappe/build issue - I did further investigations and do think it's not related to my setup. Those steps lead me to those findings:

1. Installed Frappe Docker as advices in official frappe_docker repo using pwd.yml. Did run without any issues.
2. Still no success running bench rebuild. After searching for a while, I found the solution for my errors:

> You **cannot build assets using bench build in running production containers**. It will mess up the attached assets volume for the container in which the command was executed. It will cause problems in serving static assets. The **asset building process is done during the image build.** Images come pre-packaged with built assets and there is no need to build assets in production environment.

Which means, your advice isn't possible in my case (docker setup). Bench build is no option here. I checked if there are any errors for my custom docker image built (in particular whilst building print_designer) but couldn't find so. Further, there are no problems with assets for any other app I'm currently using, therefore I suspect an Issue with print_designer in a docker environment.

I would appreciate a re-open of this issue or any reply leading me into the right direction. Thanks.
